v0.5/fastn-p1/src/parser/mod.rs

```diff
@@ -46,7 +50,15 @@ macro_rules! tt {
         #[allow(unused_macros)]
         macro_rules! t {
             ($source:expr, $debug:tt, $remaining:expr) => {
-                fastn_p1::parser::p($source, $f, serde_json::json!($debug), $remaining);
+                fastn_p1::parser::p($source, $f, serde_json::json!($debug), $remaining, None);
+            };
+        }
+        /// test with index
+        #[allow(unused_macros)]
+        macro_rules! ti {
+            ($source:expr, $debug:tt, $remaining:expr, $chars:expr, $bytes:expr) => {
+                let index = fastn_p1::parser::scanner::Index::new($chars, $bytes);
+                fastn_p1::parser::p($source, $f, serde_json::json!($debug), $remaining, Some(index));
```

Comment:
```suggestion
            /// test with index
            ($source:expr, $debug:tt, $remaining:expr, $chars:expr, $bytes:expr) => {
                let index = fastn_p1::parser::scanner::Index::new($chars, $bytes);
                fastn_p1::parser::p($source, $f, serde_json::json!($debug), $remaining, Some(index));
```

Reply:
No need to introduce a new macro name.

Reply:
Also create another clause in the macro so `remaining` is not needed, and it does an assertion that there is nothing remaining.